hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/BinaryLogicOperatorNode.java

```diff
@@ -164,14 +167,20 @@ protected void translate(
 
 		for ( int i = valueElements - 1; i > 0; i-- ) {
 			if ( i == 1 ) {
-				AST op1 = getASTFactory().create( comparisonType, comparisonText );
+				final AST op1;
+				if ( isInsideSetClause() ) {
+					op1 = container;
+				}else {
```

Comment:
super minor. we usually adopt the following code template
```
if () {
}
else {
}
```

Reply:
Fixed, 
Thanks @NathanQingyangXu